frame/core-fellowship/src/lib.rs

```diff
@@ -0,0 +1,597 @@
+		/// Member has been demoted to the given (non-zero) rank.
+		Demoted { who: T::AccountId, to_rank: RankOf<T, I> },
+		/// Member has been proven at their current rank, postponing auto-demotion.
+		Proven { who: T::AccountId, at_rank: RankOf<T, I> },
+		/// Member has stated evidence of their efforts their request for rank.
+		Requested { who: T::AccountId, wish: Wish },
+		/// Some submitted evidence was judged and removed. There may or may not have been a change
+		/// to the rank, but in any case, `last_proof` is reset.
+		EvidenceJudged {
+			/// The member/candidate.
+			who: T::AccountId,
+			/// The desired outcome for which the evidence was presented.
+			wish: Wish,
+			/// The evidence of efforts.
+			evidence: Evidence<T, I>,
+			/// The old rank, prior to this change.
+			old_rank: u16,
+			/// New rank. If `None` then candidate record was removed entirely.
+			new_rank: Option<u16>,
+		},
+		/// Pre-ranked account has been inducted at their current rank.
+		Imported { who: T::AccountId, rank: RankOf<T, I> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T, I = ()> {
+		/// Member's rank is too low.
+		Unranked,
+		/// Member's rank is not zero.
+		Ranked,
+		/// Member's rank is not as expected - generally means that the rank provided to the call
+		/// does not agree with the state of the system.
+		UnexpectedRank,
+		/// The given rank is invalid - this generally means it's not between 1 and `RANK_COUNT`.
+		InvalidRank,
+		/// The origin does not have enough permission to do this operation.
+		NoPermission,
+		/// No work needs to be done at present for this member.
+		NothingDoing,
+		/// The candidate has already been inducted. This should never happen since it would
+		/// require a candidate (rank 0) to already be tracked in the pallet.
+		AlreadyInducted,
+		/// The candidate has not been inducted, so cannot be offboarded from this pallet.
+		NotTracked,
+		/// Operation cannot be done yet since not enough time has passed.
+		TooSoon,
+	}
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Bump the state of a member.
+		///
+		/// This will demote a member whose `last_proof` is now beyond their rank's
+		/// `demotion_period`.
+		///
+		/// - `origin`: A `Signed` origin of an account.
+		/// - `who`: A member account whose state is to be updated.
+		#[pallet::weight(T::WeightInfo::bump_offboard().max(T::WeightInfo::bump_demote()))]
+		#[pallet::call_index(0)]
+		pub fn bump(origin: OriginFor<T>, who: T::AccountId) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let mut member = Member::<T, I>::get(&who).ok_or(Error::<T, I>::NotTracked)?;
+			let rank = T::Members::rank_of(&who).ok_or(Error::<T, I>::Unranked)?;
+
+			let params = Params::<T, I>::get();
+			let demotion_period = if rank == 0 {
+				params.offboard_timeout
+			} else {
+				let rank_index = Self::rank_to_index(rank).ok_or(Error::<T, I>::InvalidRank)?;
+				params.demotion_period[rank_index]
```

Comment:
just scrolling through, but feels like this should be using `get` api, which handles if the index does not exist, rather than panic. could happen if bad migrations or something else, idk